### ISSUE TITLE
Unable to configure clipboard copy (pbcopy) functionality on macOS with Wezterm

### ISSUE THREAD

**Issue by sujitawake** [sxyazi/yazi#2097]
### What system are you running Yazi on?

macOS

### What terminal are you running Yazi in?

Wezterm

### `yazi --debug` output

```Shell
Yazi
    Version: 0.4.2 (5cfcab3 2024-12-20)
    Debug  : false
    Triple : aarch64-apple-darwin (macos-aarch64)
    Rustc  : 1.83.0 (90b35a62 2024-11-26)

Ya
    Version: 0.4.2 (5cfcab3 2024-12-20)

Emulator
    Brand.from_env      : Some(WezTerm)
    Emulator.detect     : Emulator { kind: Left(WezTerm), light: false, cell_size: Some((17, 35)) }
    Emulator.detect_full: Ok(Emulator { kind: Left(WezTerm), light: false, cell_size: Some((17, 35)) })

Adapter
    Adapter.matches: Iip

Desktop
    XDG_SESSION_TYPE           : None
    WAYLAND_DISPLAY            : None
    DISPLAY                    : None
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

NVIM
    NVIM          : false
    Neovim version: 0.10.3

Variables
    SHELL           : Some("/bin/zsh")
    EDITOR          : Some("/opt/homebrew/bin/nvim")
    VISUAL          : Some("/opt/homebrew/bin/nvim")
    YAZI_FILE_ONE   : None
    YAZI_CONFIG_HOME: None
    YAZI_ZOXIDE_OPTS: None
    FZF_DEFAULT_OPTS: Some("\n--layout=reverse\n--info=inline\n--height=80%\n--multi\n--preview-window=:hidden\n--preview \'([[ -f {} ]] && (bat --style=numbers --color=always {} || cat {})) || ([[ -d {} ]] && (tree -C {} | less)) || echo {} 2> /dev/null | head -200\'\n--color=\'hl:148,hl+:154,pointer:032,marker:010,bg+:237,gutter:008\'\n--prompt=\'∼ \' --pointer=\'▶\' --marker=\'✓\'\n--bind \'?:toggle-preview\'\n--bind \'ctrl-a:select-all\'\n--bind \'ctrl-y:execute-silent(echo {+} | pbcopy)\'\n--bind \'ctrl-e:execute(echo {+} | xargs -o vim)\'\n--bind \'ctrl-v:execute(code {+})\'\n")

Text Opener
    default     : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-create: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-rename: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : 0
    tmux version       : 3.4
    tmux build flags   : enable-sixel=Unknown
    ZELLIJ_SESSION_NAME: None
    Zellij version     : 0.41.2

Dependencies
    file          : 5.41
    ueberzugpp    : No such file or directory (os error 2)
    ffmpeg/ffprobe: 7.1 / 7.1
    pdftoppm      : 24.12.0
    magick        : 7.1.1-42
    fzf           : 0.57.0
    fd/fdfind     : 10.2.0 / No such file or directory (os error 2)
    rg            : 14.1.1
    chafa         : No such file or directory (os error 2)
    zoxide        : No such file or directory (os error 2)
    7z/7zz        : 17.05 / No such file or directory (os error 2)
    jq            : 1.7.1

Clipboard
    wl-copy/paste: No such file or directory (os error 2) / No such file or directory (os error 2)
    xclip        : No such file or directory (os error 2)
    xsel         : No such file or directory (os error 2)
```

## Issue Description

Is there a recommended way to configure clipboard operations for macOS users? Or is this potentially a bug in how Yazi 0.4.2 interacts with macOS clipboard operations?

I am unable to configure Yazi to copy file contents to system clipboard using `pbcopy`. The native macOS clipboard command `pbcopy` works correctly in the terminal (`echo "test" | pbcopy` functions as expected), but cannot get it working within Yazi's configuration.

## Attempted Configurations
I've tried multiple approaches in `yazi.toml`, but none have worked:

### Attempt 1

```toml
[[manager.prepend_keymap]]
on = "y"
exec = "shell"
args = { cmd = "cat \"{}\" | pbcopy && echo 'Copied to clipboard!'" }
```

### Attempt 2

```toml
[[manager.prepend_keymap]]
on = "y"
run = [
    "shell 'cat \"$@\" | pbcopy'",
    "yank"
]
desc = "Copy file contents to clipboard"
```

### Attempt 3

```toml
[[manager.prepend_keymap]]
on = "y"
run = "shell '/bin/zsh -c \"cat \\\"$@\\\" | pbcopy\"'"
desc = "Copy file contents to clipboard"
```

### Attempt 4

```toml
[[manager.prepend_keymap]]
on = "y"
exec = "shell"
args = { command = "cat $YAZI_CURRENT_FILE | pbcopy" }
```

## Current Behavior

None of the configurations result in the file contents being copied to the clipboard. No error messages are displayed in the status bar.

### Minimal reproducer

1. Append config entries to `yazi.toml` with:

```toml
[[manager.prepend_keymap]]
on = "y"
exec = "shell"
args = { cmd = "cat \"{}\" | pbcopy" }
```

2. Create a test file:

```sh
echo "test content" > test.txt
```

3. Steps to reproduce:
  - Start yazi
  - Navigate to test.txt
  - Press 'y'
  - Try to paste in any text editor

**Expected:** The content "test content" should be in clipboard
**Actual:** Nothing is copied to clipboard

### Checklist

- [X] I tried the [latest nightly build](https://yazi-rs.github.io/docs/installation#official-binaries), and the issue is still reproducible
- [X] I updated the debug information (`yazi --debug`) input box to the nightly that I tried
- [X] I can reproduce it after disabling all custom configs/plugins (`mv ~/.config/yazi ~/.config/yazi-backup`)

**Comment by sxyazi** [comment]
I can't reproduce it with

```
[[manager.prepend_keymap]]
on = "y"
run = [
    "shell 'cat \"$@\" | pbcopy'",
    "yank"
]
```

it works fine for me

```
❯ yazi --version
Yazi 0.4.3 (d754ac5 2024-12-25)
```

**Comment by sujitawake** [comment]
Attaching a quick screencast I recorded demonstrating this behavior.

Kindly take a look.

![yazi_behavior (1)](https://github.com/user-attachments/assets/11d09b70-fc66-475e-aa8b-3ee838b35ffc)

**Comment by sujitawake** [comment]
I keep the following file `keymap.toml` in the config directory `~/.config/yazi/` having the exact content copied from the below URL:
* [keymap.toml](https://github.com/sxyazi/yazi/blob/shipped/yazi-config/preset/keymap-default.toml)

Also, here is the content of my `yazi.toml`:

```toml
[plugin]
prepend_previewers = [
    { name = "*.md", run = "glow" },
    { mime = "text/csv", run = "miller"},
    # Archive previewer
    { mime = "application/zip",           run = "ouch" },
    { mime = "application/tar",           run = "ouch" },
    { mime = "application/bzip2",         run = "ouch" },
    { mime = "application/7z-compressed", run = "ouch" },
    { mime = "application/rar",           run = "ouch" },
    { mime = "application/xz",            run = "ouch" },
]

[[manager.prepend_keymap]]
on = "y"
run = [
    "shell 'cat \"$@\" | pbcopy'",
    "yank"
]

[[manager.prepend_keymap]]
on = ["<Enter>"]
exec = "open"
args = { exec = "$EDITOR" }

append_previewers = [
    { name = "*", run = "hexyl" },
]

[preview]
wrap = "yes"

[manager]
ratio = [1,3,4] 
show_hidden = true
sort_by = "natural"         # Sort files naturally
sort_sensitive = false      # Makes sorting case-insensitive
sort_reverse = false        # Ascending order sorting
sort_dir_first = true       # Lists directories before files
scrolloff = 5               # Keeps 5 lines above and below the cursor while scrolling
show_symlink = true         # Displays symlink targets

# Better performance in file operations
[tasks]
micro_workers = 5
macro_workers = 10
bizarre_retry = 5

[plugin.fg]
use_gitignore = true
fd_command = "fd"  # Ensures fd is used for file search

[[ manager.prepend_keymap ]]
on   = [ "g","c" ]
run  = "cd ~/.config"
desc = "Go to config directory"

[[ manager.prepend_keymap ]]
on   = [ "g","h" ]
run  = "cd ~"
desc = "Go to home directory"

[[manager.prepend_keymap]]
on   = [ "g","d" ]
run  = "cd ~/Downloads"
desc = "Go to Downloads directory"

[[ manager.prepend_keymap ]]
on   = [ "g","y" ]
run  = "/Users/user/.config/yazi"
desc = "Go to Yazi directory"

[[ manager.prepend_keymap ]]
on   = [ "g","b" ]
run  = "cd /opt/homebrew/"
desc = "Go to Homebrew directory"

[[manager.prepend_keymap]]
on   = [ "f","a" ]
run  = "plugin fg --args='rga'"
desc = "Find file by content (ripgrep-all)"

[[ manager.prepend_keymap ]]
on   = [ "f","G" ]
run  = "plugin fg --args='rg'"
desc = "Find file by content recursively (ripgrep match)"

[[ manager.prepend_keymap ]]
on   = [ "f","g" ]
run  = "plugin fg"
desc = "Find file by content recursively (fuzzy match)"

[[ manager.prepend_keymap ]]
on   = [ "f","f" ]
run  = "plugin fg --args='fzf'"
desc = "Find file by filename recursively"

# Logging for troubleshooting
[log]
enabled = true
```

**Comment by hankertrix** [comment]
It looks like a configuration error.

The `yazi.toml` file you showed should not have keymaps (`manager.prepend_keymap`). All of those should go into `keymap.toml`. You shouldn't be copying the default `keymap.toml` file wholesale as well if you just want to use the default configuration.

Configuring plugins with `plugin` is also not supported yet, so I'm not sure what the `[plugin.fg]` key is supposed to do.

I also don't think there is a `log` key in the `yazi.toml` file.

I suggest adding the line below to the top of your `yazi.toml` file for [`taplo`](https://taplo.tamasfe.dev/) to show you the errors in your configuration:

```toml
"$schema" = "https://yazi-rs.github.io/schemas/yazi.json"
```

You should also read these documentation pages:
- https://yazi-rs.github.io/docs/configuration/yazi
- https://yazi-rs.github.io/docs/configuration/keymap
- https://yazi-rs.github.io/docs/configuration/theme

**Comment by sxyazi** [comment]
Ah yeah, the keybinding rules should be in `keymap.toml`, not `yazi.toml`, I think that's why it didn't take effect. 

I didn't notice that, thanks for the reminder @hankertrix .

**Comment by sujitawake** [comment]
Completely forgot to reply to this ticket yesterday.

I tried the suggestions provided by @hankertrix and everything thereafter fell into place. My configs in the wrong place was causing all the issues. Once I moved things from "yazi.toml" to "keymap.toml" followed-by fixing the TOML lint errors (found by "taplo" tool) that I had all this time, everything started working. :)

Really appreciate both of your help and quick response on this.

Kudos to @sxyazi for gifting us with this brilliant tool. Cheers!
